src/utilcode/appxutil.cpp

```diff
@@ -861,5 +861,170 @@ namespace AppX
 #endif // DACCESS_COMPILE
 } // namespace AppX
 
-
 #endif // FEATURE_CORECLR
+
+#if defined(FEATURE_DELAYLOAD_WINRT) && !defined(DACCESS_COMPILE)
+// Helpers for various WinRT functions that need to be delay loaded for CoreCLR
+// since they are not present on all supported platforms.
+
+#include "delayloadhelpers.h"
+#include "windowsruntime.h"
+
+// Declare the list of functions to be delay-loaded
+DELAY_LOADED_MODULE("api-ms-win-core-winrt-string-l1-1-0", API_MS_WIN_CORE_WINRT_STRING_L1_1_0)
+
+DELAY_LOADED_FUNCTION_BEGIN_LIST(API_MS_WIN_CORE_WINRT_STRING_L1_1_0)
+DELAY_LOADED_FUNCTION_EX(API_MS_WIN_CORE_WINRT_STRING_L1_1_0, WindowsCreateString)
+DELAY_LOADED_FUNCTION_EX(API_MS_WIN_CORE_WINRT_STRING_L1_1_0, WindowsGetStringRawBuffer)
+DELAY_LOADED_FUNCTION_EX(API_MS_WIN_CORE_WINRT_STRING_L1_1_0, WindowsCreateStringReference)
+DELAY_LOADED_FUNCTION_EX(API_MS_WIN_CORE_WINRT_STRING_L1_1_0, WindowsDeleteString)
+DELAY_LOADED_FUNCTION_EX(API_MS_WIN_CORE_WINRT_STRING_L1_1_0, WindowsGetStringLen)
+DELAY_LOADED_FUNCTION_END_LIST(API_MS_WIN_CORE_WINRT_STRING_L1_1_0)
+
+DELAY_LOADED_MODULE("api-ms-win-core-winrt-l1-1-0", API_MS_WIN_CORE_WINRT_L1_1_0)
+
+DELAY_LOADED_FUNCTION_BEGIN_LIST(API_MS_WIN_CORE_WINRT_L1_1_0)
+DELAY_LOADED_FUNCTION_EX(API_MS_WIN_CORE_WINRT_L1_1_0, RoGetActivationFactory)
+DELAY_LOADED_FUNCTION_EX(API_MS_WIN_CORE_WINRT_L1_1_0, RoUninitialize)
+DELAY_LOADED_FUNCTION_EX(API_MS_WIN_CORE_WINRT_L1_1_0, RoInitialize)
+DELAY_LOADED_FUNCTION_END_LIST(API_MS_WIN_CORE_WINRT_L1_1_0)
+
+DELAY_LOADED_MODULE("api-ms-win-core-winrt-roparameterizediid-l1-1-0", API_MS_WIN_CORE_WINRT_ROPARAMETERIZEDIID_L1_1_0)
+
+DELAY_LOADED_FUNCTION_BEGIN_LIST(API_MS_WIN_CORE_WINRT_ROPARAMETERIZEDIID_L1_1_0)
+DELAY_LOADED_FUNCTION_EX(API_MS_WIN_CORE_WINRT_ROPARAMETERIZEDIID_L1_1_0, RoGetParameterizedTypeInstanceIID)
+DELAY_LOADED_FUNCTION_END_LIST(API_MS_WIN_CORE_WINRT_ROPARAMETERIZEDIID_L1_1_0)
+
+DELAY_LOADED_MODULE("api-ms-win-ro-typeresolution-l1-1-0", API_MS_WIN_RO_TYPERESOLUTION_L1_1_0)
+
+DELAY_LOADED_FUNCTION_BEGIN_LIST(API_MS_WIN_RO_TYPERESOLUTION_L1_1_0)
+DELAY_LOADED_FUNCTION_EX(API_MS_WIN_RO_TYPERESOLUTION_L1_1_0, RoParseTypeName)
+DELAY_LOADED_FUNCTION_END_LIST(API_MS_WIN_RO_TYPERESOLUTION_L1_1_0)
+
+HRESULT WINAPI WindowsCreateString(LPCWSTR sourceString, UINT32 length, HSTRING *string)
+{
+    HRESULT hr = S_OK;
+
+    typedef HRESULT WINAPI pFuncWindowsCreateString(LPCWSTR sourceString, UINT32 length, HSTRING *string);
+
+    pFuncWindowsCreateString *pWindowsCreateString = nullptr;
+    IfFailThrow(DelayLoad::API_MS_WIN_CORE_WINRT_STRING_L1_1_0::WindowsCreateString.GetValue(&pWindowsCreateString));
```

Comment:
I agree that failing fast for not being able to find these API imports is a better thing to do since the caller cannot do anything reasonable if the API is not found (and callsites of these APIs usually thrown an exception is WinRT is not supported).